azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/EventPosition.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+package com.microsoft.azure.eventhubs;
+
+import com.microsoft.azure.eventhubs.amqp.AmqpConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+
+/**
+ * Defines a position of an {@link EventData} in the event hub partition.
+ * The position can be an Offset, Sequence Number, or EnqueuedTime.
+ */
+public class EventPosition {
+
+    private static final Logger TRACE_LOGGER = LoggerFactory.getLogger(EventPosition.class);
+
+    private final String offset;
+    private final Long sequenceNumber;
+    private final Instant dateTime;
+    private final Boolean inclusiveFlag;
+
+    private EventPosition(String o, Long s, Instant d, Boolean i) {
+        this.offset = o;
+        this.sequenceNumber = s;
+        this.dateTime = d;
+        this.inclusiveFlag = i;
+    }
+
+    /**
+     * Creates a position at the given offset. The specified event will not be included.
+     * Instead, the next event is returned.
+     * @param offset    is the byte offset of the event.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromOffset(String offset) {
+        return EventPosition.fromOffset(offset, false);
+    }
+
+    /**
+     * Creates a position at the given offset.
+     * @param offset    is the byte offset of the event.
+     * @param inclusiveFlag will include the specified event when set to true; otherwise, the next event is returned.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromOffset(String offset, boolean inclusiveFlag) {
+        return new EventPosition(offset, null, null, inclusiveFlag);
+    }
+
+    /**
+     * Creates a position at the given sequence number. The specified event will not be included.
+     * Instead, the next event is returned.
+     * @param sequenceNumber is the sequence number of the event.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromSequenceNumber(Long sequenceNumber) {
+        return EventPosition.fromSequenceNumber(sequenceNumber, false);
+    }
+
+    /**
+     * Creates a position at the given sequence number. The specified event will not be included.
+     * Instead, the next event is returned.
+     * @param sequenceNumber    is the sequence number of the event.
+     * @param inclusiveFlag         will include the specified event when set to true; otherwise, the next event is returned.
+     * @return An {@link EventPosition} object.
+     */
+    public static EventPosition fromSequenceNumber(Long sequenceNumber, boolean inclusiveFlag) {
+        return new EventPosition(null, sequenceNumber, null, inclusiveFlag);
+    }
```

Comment:
We don't need this overload since inclusiveFlag is only used on the offset.

Reply:
We'll have inclusiveFlags for seqNumbers as well.